test/suites/azuregarbagecollection/suite_test.go

```diff
@@ -0,0 +1,148 @@
+		vnetResourceGroup := os.Getenv("VNET_RESOURCE_GROUP")
+		if vnetResourceGroup == "" {
+			vnetResourceGroup = resourceGroup
+		}
+		interfacesClient, vnetClient, err := newAzureClients(subscriptionID)
+		Expect(err).ToNot(HaveOccurred())
+		vnet, err := getVNET(env.Context, vnetClient, vnetResourceGroup)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = createOrphanNIC(env.Context, interfacesClient, resourceGroup, env.Region, vnet.Properties.Subnets[0])
+		Expect(err).ToNot(HaveOccurred())
+		EventuallyExpectOrphanNicsToBeDeleted(env, resourceGroup, interfacesClient)
+	})
+})
+
+func newAzureClients(subscriptionID string) (*armnetwork.InterfacesClient, *armnetwork.VirtualNetworksClient, error) {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	interfacesClient, err := armnetwork.NewInterfacesClient(subscriptionID, cred, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	vnetClient, err := armnetwork.NewVirtualNetworksClient(subscriptionID, cred, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	return interfacesClient, vnetClient, nil
+}
+
+func getVNET(ctx context.Context, client *armnetwork.VirtualNetworksClient, vnetRG string) (*armnetwork.VirtualNetwork, error) {
+	pager := client.NewListPager(vnetRG, nil)
+	for pager.More() {
+		resp, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list virtual networks: %w", err)
+		}
+		if len(resp.VirtualNetworkListResult.Value) > 0 {
+			return resp.VirtualNetworkListResult.Value[0], nil
+		}
+	}
+	return nil, fmt.Errorf("no virtual networks found in resource group: %s", vnetRG)
+}
+
+func createOrphanNIC(ctx context.Context, client *armnetwork.InterfacesClient, resourceGroup, region string, subnet *armnetwork.Subnet) error {
+	nodepoolKey := strings.ReplaceAll(karpv1.NodePoolLabelKey, "/", "_")
+	poller, err := client.BeginCreateOrUpdate(ctx, resourceGroup, nicName, armnetwork.Interface{
+		Location: to.Ptr(region),
+		Tags:     map[string]*string{nodepoolKey: lo.ToPtr("default")},
+		Properties: &armnetwork.InterfacePropertiesFormat{
+			IPConfigurations: []*armnetwork.InterfaceIPConfiguration{
+				{
+					Name: lo.ToPtr(nicName),
+					Properties: &armnetwork.InterfaceIPConfigurationPropertiesFormat{
+						Primary:                   lo.ToPtr(true),
+						Subnet:                    subnet,
+						PrivateIPAllocationMethod: lo.ToPtr(armnetwork.IPAllocationMethodDynamic),
+					},
+				},
+			},
+		},
+	}, nil)
+	if err != nil {
+		return err
+	}
+	_, err = poller.PollUntilDone(ctx, nil)
+	return err
+}
+
+func EventuallyExpectOrphanNicsToBeDeleted(env *azure.Environment, resourceGroup string, nicClient *armnetwork.InterfacesClient) {
+	GinkgoHelper()
+	Eventually(func() bool {
+		_, err := nicClient.Get(env.Context, resourceGroup, nicName, nil)
+		var respErr *azcore.ResponseError
+		if errors.As(err, &respErr) && respErr.StatusCode == 404 {
+			return true
```

Comment:
Can we please use a shared helper (even just within this code base) so that the checks for not found all use the same logic?